### PR TITLE
Fire event for when plugin integration form is saved

### PR DESCRIPTION
**Pull request by alanhartless**
Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | n
| New feature?  | y 
| BC breaks?    | n
| Deprecations? | n
| Fixed issues  |  

## Description

This dispatches an event when a plugin integration config is saved to allow listeners to execute tasks accordingly.

It also fixes a UI issue with the config form when no features are enabled, only requires keys when the integration is published, and prevents a couple PHP notices found while testing this.

## Steps to test this PR

This is mainly developer oriented so apply the PR, delete cache and ensure a integration's config can still be saved.

### CODE REVIEW

File: app/bundles/PluginBundle/Integration/AbstractIntegration.php [hunk 1]
```diff
@@ -1570,6 +1580,7 @@ public function logIntegrationError(\Exception $e)
     {
         $logger = $this->factory->getLogger();
         $logger->addError('INTEGRATION ERROR: '.$this->getName().' - '.$e->getMessage());
+        die(var_dump($e));
```
Comment (escopecz) [inline app/bundles/PluginBundle/Integration/AbstractIntegration.php]: Oops ;)

Comment (alanhartless) [inline app/bundles/PluginBundle/Integration/AbstractIntegration.php]: Pesky little buggers. :-)

**Comment by escopecz** [comment]
Plugin configuration is working for me :+1:

**Comment by dongilbert** [comment]
+1

**Pull request merged**